libs/order/driver/magento/2.4.1/src/transforms/responses/order.ts

```diff
@@ -0,0 +1,280 @@
+import { daffAdd, daffSubtract } from '@daffodil/core';
+import {
+  DaffOrder,
+  DaffOrderTotal,
+  DaffOrderItem,
+  DaffOrderAddress,
+  DaffOrderShipment,
+  DaffOrderShipmentItem,
+  DaffOrderShipmentTracking,
+  DaffOrderPayment,
+  DaffOrderInvoice,
+  DaffOrderTotalTypeEnum,
+  DaffCompositeOrderItemOption,
+  DaffOrderItemType,
+  DaffConfigurableOrderItemAttribute,
+  DaffOrderCoupon
+} from '@daffodil/order';
+
+import {
+  MagentoOrder,
+  MagentoOrderItem,
+  MagentoOrderShipment,
+  MagentoOrderAddress,
+  MagentoOrderShipmentItem,
+  MagentoOrderShipmentTracking,
+  MagentoOrderPayment,
+  MagentoOrderInvoice
+} from '../../models/responses/public_api';
+import { MagentoOrderTotal } from '../../models/responses/order-total';
+import { MagentoOrderCredit } from '../../models/responses/order-credit';
+import { MagentoOrderCreditItem } from '../../models/responses/order-credit-item';
+import { MagentoOrderBundleItem, MagentoOrderBundleItemSelectedOption, MagentoOrderItemOption, MagentoOrderItemType } from '../../models/responses/order-item';
+import { MagentoDiscount } from '../../models/responses/discount';
+import { MagentoOrderInvoiceItem } from '../../models/responses/order-invoice-item';
+
+function transformTotals(totals: MagentoOrderTotal): DaffOrderTotal[] {
+  return [
+    {
+      label: 'Grand Total',
+      type: DaffOrderTotalTypeEnum.GrandTotal,
+      value: totals.grand_total.value,
+      sort_order: 1
+    },
+    {
+      label: 'Subtotal',
+      type: DaffOrderTotalTypeEnum.Subtotal,
+      value: totals.subtotal.value,
+      sort_order: 0
+    },
+    {
+      label: 'Shipping',
+      type: DaffOrderTotalTypeEnum.Shipping,
+      value: totals.total_shipping.value,
+      sort_order: 2
+    },
+    {
+      label: 'Tax',
+      type: DaffOrderTotalTypeEnum.Tax,
+      value: totals.total_tax.value,
+      sort_order: 3
+    },
+    {
+      label: 'Discount',
+      type: DaffOrderTotalTypeEnum.Discount,
+      value: totals.discounts.reduce((acc, discount) => daffAdd(acc, discount.amount.value), 0),
+      sort_order: 4
+    }
+  ]
+}
+
+function transformCouponDiscount(discount: MagentoDiscount): DaffOrderCoupon {
+  return {
+    code: discount.label,
+  }
+}
+
+function transformConfigurableOption(option: MagentoOrderItemOption): DaffConfigurableOrderItemAttribute {
+  return {
+    attribute_label: option.label,
+	  value_label: option.value
+  }
+}
+
+function transformBundleOption(option: MagentoOrderBundleItemSelectedOption): DaffCompositeOrderItemOption {
+  return {
+    option_label: option.label,
+	  value_label: option.values[0].product_name
```

Comment:
Needs a null check?

Reply:
I haven't seen that `values` can ever be `null` but the schema doesn't prevent it so I'll add it to be safe. added